Upgrade updates (#133)
mulesoft/docs-runtime-fabric: modules/ROOT/pages/upgrade-cluster.adoc

= Perform Runtime Fabric Cluster Upgrades

Anypoint Runtime Fabric is powered by a set of components including Docker, Kubernetes, and the Ops Center. These components run on virtual machines or bare-metal servers that act as nodes on the cluster. Perform the following steps to upgrade these cluster-level components and apply updates to node configurations.

== Before You Begin

* Run the upgrade procedure on a non-production Runtime Fabric to verify compatibility with your environment and to gain familiarity with the procedure.
* Plan the upgrade procedure when your team does not need to deploy new applications or managing existing applications on Runtime Fabric.

** The upgrade can take over an hour to complete.

> ** The connection between Runtime Fabric and Anypoint Runtime Manager might become disconnected multiple times during the upgrade procedure. Connectivity will be restored after the upgrade has completed.

Reviewer: new style guidance says "no" to modal verbs: https://confluence.internal.salesforce.com/pages/releaseview.action?spaceKey=DOCTEAM&title=modal+verbs

Author: Fixed - thank you!

> ** The connection between Runtime Fabric and Anypoint Runtime Manager might become disconnected multiple times during the upgrade procedure. Connectivity will be restored after the upgrade has completed.

Reviewer: ```suggestion
** The connection between Runtime Fabric and Anypoint Runtime Manager might become disconnected multiple times during the upgrade procedure. Connectivity is restored after the upgrade completes.
```

Author: Got it - thank you!

* Make sure the `/tmp` directory has a minimum of 4 GiB of disk space available.
* Verify you have root privileges. Root privileges are required to perform an upgrade.

=== Environment Prerequisites

* A production configuration of Runtime Fabric, with a minimum of 3 controller VMs, is required.
* The internal load balancer must run on at least 3 replicas to maintain availability.
* An external load balancer must be configured to load balance incoming requests to the controller VMs, with a health check configured for TCP port 443.

> * At least one worker node’s worth of CPU and memory resources must be available in the cluster.

Reviewer: Is there a way to reword?

Author: Leaving for now.

+
This enables safe removal of a worker node from the cluster to apply upgrades, without impacting availability of applications.
* Applications serving inbound requests must be scaled to a minimum of 2 replicas.

> [WARNING] Runtime Fabrics running on cluster version 1.0.x encounter downtime to applications if configured to use an HTTP proxy. An additional step in the procedure is required to apply the HTTP proxy configuration on each node after the upgrade. The Runtime Fabric version can be found by running `gravity status` on a node and referencing the cluster version.

Reviewer: any way to reword?

Author: Leaving for now.

> [WARNING] Runtime Fabrics running on cluster version 1.0.x encounter downtime to applications if configured to use an HTTP proxy. An additional step in the procedure is required to apply the HTTP proxy configuration on each node after the upgrade. The Runtime Fabric version can be found by running `gravity status` on a node and referencing the cluster version.

Reviewer: Reword to be active voice?

Author: Reworded a bit - thank you!

sudo ./gravity upgrade --phase=< insert phase > --force --debug
----
+

> For example,  `--phase=/gc/rtf-controller-1`

Reviewer: terminating punctuation?

Author: Added that - thank you!

. Wait for the command to run again. If the command does not terminate with an error, resume the upgrade by running the following command:
+
----

> `sudo ./gravity upgrade --resume`

Reviewer 2: Remove the first and last ticks

Author: Got it - thank you!

sudo ./gravity plan execute --phase=< insert phase > --force --debug
----

> The command should complete successfully. If an error is returned, wait a few minutes and repeat the previous steps.

Reviewer: modal :)

Author: Fixed - thank you!
